$MirOS: ports/www/firesomething/patches/patch-nsprpub_configure_in,v 1.6 2008/05/21 06:31:04 bsiegert Exp $
$OpenBSD: patch-nsprpub_configure_in,v 1.3 2005/04/09 15:43:47 wilfried Exp $
--- nsprpub/configure.in.orig	2008-04-29 01:21:11.000000000 +0200
+++ nsprpub/configure.in	2008-08-24 14:52:51.000000000 +0200
@@ -42,7 +42,7 @@ dnl ***** END LICENSE BLOCK *****
 AC_PREREQ(2.12)
 AC_INIT(config/libc_r.h)
 
-AC_CONFIG_AUX_DIR(${srcdir}/build/autoconf)
+AC_CONFIG_AUX_DIR(build/autoconf)
 AC_CANONICAL_SYSTEM
 
 dnl ========================================================
@@ -659,6 +659,16 @@ else
     OS_TEST=`uname -m`
 fi
 
+if test "$OS_ARCH" = "MidnightBSD"; then
+    OS_ARCH="FreeBSD"
+    OS_RELEASE="6.0" # XXX
+fi
+
+if test "$OS_ARCH" = "MirBSD"; then
+    OS_ARCH="OpenBSD"
+    OS_RELEASE="3.9" # XXX
+fi
+
 if test "$OS_ARCH" = "IRIX64"; then
     OS_ARCH=IRIX
 fi
@@ -1797,14 +1807,34 @@ mips-sony-newsos*)
     _DEBUG_FLAGS="-gstabs"
 	;;
 
-*-openbsd*)
+*-midnightbsd*)
+    if test -z "$USE_NSPR_THREADS"; then
+        USE_PTHREADS=1
+    fi
+    AC_DEFINE(XP_UNIX)
+    AC_DEFINE(FREEBSD)
+    AC_DEFINE(HAVE_BSD_FLOCK)
+    AC_DEFINE(HAVE_SOCKLEN_T)
+    CFLAGS="$CFLAGS $(DSO_CFLAGS) -ansi -Wall"
+    MOZ_OBJFORMAT=elf
+    DLL_SUFFIX="so"
+    MKSHLIB='$(CC) $(DSO_LDOPTS) -o $@'
+    DSO_CFLAGS=-fPIC
+    DSO_LDOPTS='-shared -Wl,-soname -Wl,$(notdir $@)'
+    MDCPUCFG_H=_freebsd.cfg
+    PR_MD_CSRCS=freebsd.c
+    ;;
+
+*-openbsd*|*-mirbsd*)
     AC_DEFINE(XP_UNIX)
     AC_DEFINE(OPENBSD)
     AC_DEFINE(HAVE_BSD_FLOCK)
     AC_DEFINE(HAVE_SOCKLEN_T)
+    AC_DEFINE(_PR_HAVE_GETPROTO_R)
+    AC_DEFINE(_PR_HAVE_GETPROTO_R_INT)
     CFLAGS="$CFLAGS -ansi -Wall"
     CXXFLAGS="$CXXFLAGS -ansi -Wall"
-    DLL_SUFFIX=so.1.0
+    DLL_SUFFIX="so.${SO_VERSION}"
     DSO_CFLAGS=-fPIC
     MDCPUCFG_H=_openbsd.cfg
     PR_MD_CSRCS=openbsd.c
@@ -2528,7 +2558,7 @@ if test -n "$USE_PTHREADS"; then
             _PTHREAD_LDFLAGS=
         fi
 	    ;;
-    *-freebsd*)
+    *-freebsd*|*-midnightbsd*)
 	    AC_DEFINE(_REENTRANT)
 	    AC_DEFINE(_THREAD_SAFE)
 	    dnl -pthread links in -lc_r, so don't specify it explicitly.
@@ -2550,7 +2580,7 @@ if test -n "$USE_PTHREADS"; then
 	        _PTHREAD_LDFLAGS=
 	    fi
 	    ;;
-    *-openbsd*)
+    *-openbsd*|*-mirbsd*)
         if test "$ac_cv_have_dash_pthread" = "yes"; then
             _PTHREAD_LDFLAGS=-pthread
         fi
@@ -2650,7 +2680,7 @@ case "$target" in
     _PTHREAD_LDFLAGS=
     USE_USER_PTHREADS=
     ;;
-*-netbsd*|*-openbsd*)
+*-netbsd*|*-openbsd*|*-mirbsd*|*-midnightbsd*)
     if test -n "$USE_NSPR_THREADS"; then
         AC_DEFINE(_PR_LOCAL_THREADS_ONLY)
     fi
